traitlets/traitlets.py

```diff
@@ -593,6 +637,35 @@ class dict to the newly created class ``cls``.
         super(MetaHasTraits, cls).__init__(name, bases, classdict)
 
 
+def observe(*names):
+    """ A decorator which can be used to observe members on a class.
+
+    Parameters
+    ----------
+    *names
+        The str names of the attributes to observe on the object.
+    """
+    return ObserveHandler(names)
+
+
+class ObserveHandler(BaseDescriptor):
+
+    def __init__(self, names=None):
+        if names is None:
+            self.names=[None]
+        else:
+            self.names = names
+
+    def __call__(self, func):
+        self.func = func
+        return self
+
+    def instance_init(self, inst):
```

Comment:
It turns out that decorators prevent functions from being registered as methods at run time. `instance_init` should instead read:

```python
    def instance_init(self, inst):
        self.func = types.MethodType(self.func, inst)
        for name in self.names:
            inst.observe(self.func, name= name)